### PR TITLE
net, netstat: Remove pod network iface name normalization

### PR DESCRIPTION
**Pull request by EdDev**
**What this PR does / why we need it**:

Back in https://github.com/kubevirt/kubevirt/pull/1595 (from 2018), a
normalization has been added to populate the pod network iface name in
case it was missing, for compatibility reasons.

Since, another restriction was added: Only when the domain is not
reporting this iface as part of it.

As the normalization is performed now for more than 3 years and such old
virt-launchers are not expected to be supported by current
virt-handlers, this should be safer to remove.

The comment on the API, about making the iface name mandatory is also
removed, as this is known to be a valid option. Ifaces detected in the
guest by the guest-agent, with no ability to link them to an existing
network in the VMI spec will miss the name.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
~~Depends on #6781 , please review here only the last commit in the chain.~~

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-operator/resource/generate/components/validations_generated.go [hunk 1]
```diff
@@ -8440,9 +8440,8 @@ var CRDsValidation map[string]string = map[string]string{
                 description: Hardware address of a Virtual Machine interface
                 type: string
               name:
-                description: 'Name of the interface, corresponds to name of the network
-                  assigned to the interface TODO: remove omitempty, when api breaking
-                  changes are allowed'
+                description: Name of the interface, corresponds to name of the network
+                  assigned to the interface
```
Comment (maiqueb) [inline pkg/virt-operator/resource/generate/components/validations_generated.go:8444]: I don't understand this part. AFAIU, the pod's interface name still has the `omitEmpty` property: https://github.com/kubevirt/api/blob/f1689bb1fb9d9be9ea42bbe0efc9b29fa94ee9f8/core/v1/types.go#L501

**Review by maiqueb — CHANGES_REQUESTED** [review]
I don't follow half these changes, but I support the other one.

File: pkg/virt-operator/resource/generate/components/validations_generated.go (hunk 1 above)
Comment (EdDev) [reply pkg/virt-operator/resource/generate/components/validations_generated.go:8444]: I tried to explain it in the commit message. Perhaps you can help rephrase it to be clearer.
The TODO is wrong, it should be optional and not mandatory. One such scenario was presented there.

File: pkg/network/setup/netstat.go [hunk 2]
```diff
@@ -66,16 +66,6 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 		return nil
 	}
 
-	// This is needed to be backwards compatible with vmi's which have status interfaces
-	// with the name not being set
-	if len(domain.Spec.Devices.Interfaces) == 0 && len(vmi.Status.Interfaces) == 1 && vmi.Status.Interfaces[0].Name == "" {
```
Comment (maiqueb) [inline pkg/network/setup/netstat.go]: Now that I think about this a bit more, this is puzzling (to me at least). 

How can the VM be up n' running **but** the domain fail to report the interface whose connectivity was extended from the pod interface ?  ( or any, for all it matters)

Comment (EdDev) [reply pkg/network/setup/netstat.go]: Are you trying to convince me that this code is no longer needed? :)

In theory, this state may exist, just because someone detached the interfaces from the domain under the hood.

IMO this is just a leftover dead code, but to be on the safe side, the reasoning should also include its irrelevance compared to its original intention.

Comment (maiqueb) [reply pkg/network/setup/netstat.go]: > Are you trying to convince me that this code is no longer needed? :)

0_o ... I'm ... not sure.

I think you're right . This code is not needed - at least I can't understand how this scenario can come to exist .

> 
> In theory, this state may exist, just because someone detached the interfaces from the domain under the hood.
> 
> IMO this is just a leftover dead code, but to be on the safe side, the reasoning should also include its irrelevance compared to its original intention.

Whatever you prefer.

**Comment by EdDev** [comment]
change: Rebase over #6781

**Review by maiqueb — APPROVED** [review]

**Comment by EdDev** [comment]
change: Rebase

File: staging/src/kubevirt.io/api/core/v1/types.go [hunk 3]
```diff
@@ -501,7 +501,6 @@ type VirtualMachineInstanceNetworkInterface struct {
 	// Hardware address of a Virtual Machine interface
 	MAC string `json:"mac,omitempty"`
 	// Name of the interface, corresponds to name of the network assigned to the interface
-	// TODO: remove omitempty, when api breaking changes are allowed
```
Comment (AlonaKaplan) [inline staging/src/kubevirt.io/api/core/v1/types.go:504]: > The comment on the API, about making the iface name mandatory is also
> removed, as this is known to be a valid option. Ifaces detected in the
> guest by the guest-agent, with no ability to link them to an existing
> network in the VMI spec will miss the name.

AFAIU the [code](https://github.com/kubevirt/kubevirt/blob/fc60a8edc53665350b6aac5efcd69eeb2200e40d/pkg/virt-handler/vm.go#L974), even those interfaces will have a name.
